test/chaat/handler/handler_test.clj

```diff
@@ -0,0 +1,77 @@
+(ns chaat.handler.handler-test
+  (:require [clojure.test :refer :all]
+            [chaat.handler.handler :as handler]
+            [java-time.api :as jt]
+            [chaat.model.user :as model.user]
+            [chaat.fixture :as fixture]))
+
+(use-fixtures :each fixture/test-fixture)
+
+(deftest health-check-test
+  (testing "Request to health check endpoint should return status 200"
+    (let [datasource (:db fixture/test-system)
+          request {}
+          response (handler/health-check datasource request)]
+      (is (= 200 (:status response))))))
+
+(deftest signup-test
+  (let [datasource (:db fixture/test-system)]
+    (testing "Missing username or password in request params, signup fails"
+      (let [params {:password "12345678"}
+            request {:params params}
+            response (handler/signup datasource request)]
+        (is (= 400 (:status response)))))
+
+    (testing "Nil username or password in request params, signup fails"
+      (let [params {:username "john"
+                    :password nil}
+            request {:params params}
+            response (handler/signup datasource request)]
+        (is (= 400 (:status response)))))
+
+    (testing "Username and password both present in request params, signup succeeds"
+      (let [params {:username "john"
+                    :password "12345678"}
+            request {:params params}
+            response (handler/signup datasource request)]
+        (is (= 200 (:status response)))))
+
+      ;; perhaps no need to test this here? been tested at other layers already.
+      ;; depends on previous state, move to different block?
+    (testing "Username and password both present, 
+                but user already exists, hence signup fails"
+      (let [params {:username "john"
+                    :password "12345678"}
+            request {:params params}
+            response (handler/signup datasource request)]
+        (is (= 400 (:status response)))))))
+
+(deftest delete-user-test
+  (let [datasource (:db fixture/test-system)]
+    (testing "Missing username in request, delete fails"
+      (let [params {}
+            request {:params params}
+            response (handler/delete-user datasource request)]
+        (is (= 400 (:status response)))))
+
+    (testing "Username param in request with value nil, delete fails"
+      (let [params {:username nil}
+            request {:params params}
+            response (handler/delete-user datasource request)]
+        (is (= 400 (:status response)))))
+
+    ;; perhaps no need, already tested in other layers?
```

Comment:
Can you check if this should be returning 400? The delete hasn't necessarily failed if the resource doesn't exist

Reply:
Hmm, same situation as the previous signup case. Will post about this too.

Reply:
Will use 404 not found.